misc/openapi-ballerina/modules/openapi-validator/src/main/java/org/ballerinalang/openapi/validator/ValidatorUtil.java

```diff
@@ -281,29 +290,44 @@ static void validateResourcesAgainstOpenApi(List<String> tags, List<String> oper
                                         if (schema != null) {
                                             isExist = validateResourceAgainstOpenAPIParams(parameter,
                                                     parameter.getParameter().symbol, schema, dLog, resourceMethod,
-                                                    resourceSummary.getPath());
+                                                    resourceSummary.getPath(), failOnErrors);
                                         }
                                     }
                                 } else if (openAPIParameter.getName().equals(parameter.getName())) {
                                     isExist = validateResourceAgainstOpenAPIParams(parameter,
                                             parameter.getParameter().symbol,
                                             openAPIParameter.getParameter().getSchema(), dLog, resourceMethod,
-                                            resourceSummary.getPath());
+                                            resourceSummary.getPath(), failOnErrors);
                                 }
                             }
 
                             if (!isExist) {
-                                dLog.logDiagnostic(Diagnostic.Kind.ERROR, parameter.getParameter().getPosition(),
-                                        ErrorMessages.undocumentedResourceParameter(parameter.getName(),
-                                                resourceMethod, resourceSummary.getPath()));
+                                if (failOnErrors) {
+                                    dLog.logDiagnostic(Diagnostic.Kind.ERROR, parameter.getParameter().getPosition(),
```

Comment:
Instead of having an if condition y not passing Diagnostic.Kind.ERROR using a variable.